src/test/java/com/emc/object/s3/V2STSBasicTest.java

```diff
@@ -0,0 +1,711 @@
+        URL url = tempClient.getPresignedUrl("test-bucket", "解析依頼C1B068.txt", new Date(1500998758000L));
+        Assert.assertEquals("http://10.246.153.111:9020/test-bucket/%E8%A7%A3%E6%9E%90%E4%BE%9D%E9%A0%BCC1B068.txt?" +
+                        "AWSAccessKeyId=ASIAB51133607AA785B5&Expires=1500998758" +
+                        "&Signature=VT%2B3Yl9Vbg8MU%2FIvLrpK1H%2BLfUI%3D" +
+                        "&X-Amz-Security-Token=" + SESSION_TOKEN,
+                url.toString());
+    }
+
+    @Test
+    public void testPreSignedUrlWithHeaders() throws Exception {
+        S3Client tempClient = getPresignDummyClient();
+        URL url = tempClient.getPresignedUrl(
+                new PresignedUrlRequest(
+                        Method.PUT, "johnsmith", "photos/puppy.jpg", new Date(1175139620000L))
+                        .withObjectMetadata(
+                                new S3ObjectMetadata().withContentType("image/jpeg")
+                                        .withContentMd5("4gJE4saaMU4BqNR0kLY+lw==")));
+        Assert.assertEquals("http://10.246.153.111:9020/johnsmith/photos/puppy.jpg" +
+                        "?AWSAccessKeyId=ASIAB51133607AA785B5&Expires=1175139620" +
+                        "&Signature=CINGoX001if%2Bz18UBYD6casmBSY%3D" +
+                        "&X-Amz-Security-Token=" + SESSION_TOKEN,
+                url.toString());
+    }
+
+    @Test
+    public void testPreSignedUrlHeaderOverrides() throws Exception {
+        super.testPreSignedUrlHeaderOverrides();
+    }
+
+    @Test
+    public void testVPoolHeader() throws Exception {
+        super.testVPoolHeader();
+    }
+
+    /**
+     * A debugging proxy (Fiddler, Charles), is required to verify that the proper header is being sent.
+     * Optionally a jersey filter could be used to sniff for it
+     */
+    @Test
+    public void testCustomHeader() {
+        super.testCustomHeader();
+    }
+
+    @Test
+    public void testStaleReadsAllowed() {
+        super.testStaleReadsAllowed();
+    }
+
+    @Test
+    public void testListMarkerWithSpecialChars() {
+        super.testListMarkerWithSpecialChars();
+    }
+
+    @Test
+    public void testListPagesNoDelimiter() {
+        super.testListPagesNoDelimiter();
+    }
+
+    @Test
+    public void testListMarkerWithIllegalChars() {
+        super.testListMarkerWithIllegalChars();
+    }
+
+    @Test
+    public void testPing() {
+        super.testPing();
+    }
+
+    @Test
+    public void testTimeouts() throws Exception {
+        super.testTimeouts();
+    }
+
+    @Test
+    public void testFaultInjection() throws Exception {
+        super.testFaultInjection();
+    }
+
+    @Test
+    public void testCifsEcs() {
+        super.testCifsEcs();
+    }
+
+    @Test
+    public void testListBuckets() throws Exception {
+        super.testListBuckets();
+    }
+
+    @Ignore
+    @Test
+    public void testMultipleVdcs() throws Exception {
+    }
+
+    @Ignore
```

Comment:
why are these ignored for STS?

Reply:
testMultipleVdcs(),testMpuAbortInMiddle() are because of the request for VDCs, is implemented in smartclient, which has not supported sts yet
the others are banned for STS user by design